src/Mailer/Mailer.php

```diff
@@ -69,17 +93,39 @@ public function sendResettingEmailMessage(UserInterface $user): void
         $renderedLines = preg_split('/\R/', trim($rendered), 2, PREG_SPLIT_NO_EMPTY);
         $subject = array_shift($renderedLines);
         $body = implode('', $renderedLines);
-        $message = (new \Swift_Message())
-            ->setSubject($subject)
-            ->setFrom($this->fromEmail)
-            ->setTo((string) $user->getEmail())
-            ->setBody($body);
 
-        $this->mailer->send($message);
+        // NEXT_MAJOR: Remove this condition.
+        if ($this->mailer instanceof \Swift_Mailer) {
+            $this->sendResettingEmailMessageWithSwiftMailer($user, $subject, $body);
+
+            return;
+        }
+
+        $this->mailer->send(
+            (new Email())
```

Comment:
I think it will not possible in a transparent way, since the Twig template is currently providing the subject and the body:
```php
$renderedLines = preg_split('/\R/', trim($rendered), 2, PREG_SPLIT_NO_EMPTY);
        $subject = array_shift($renderedLines);
        $body = implode('', $renderedLines);
```